src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -809,6 +835,20 @@ internal BoundExpression ConvertPatternExpression(
             }
 
             constantValue = convertedExpression.ConstantValueOpt;
+
+            if (unionType is not null && !convertedExpression.HasErrors && constantValue is { IsBad: false } && convertedExpression.Type is not null)
+            {
+                CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = GetNewCompoundUseSiteInfo(diagnostics);
+                ConstantValue match = ExpressionOfTypeMatchesUnionPatternType(Conversions, unionType, convertedExpression.Type, ref useSiteInfo, out _, operandConstantValue: null);
+                if (match == ConstantValue.False || match == ConstantValue.Bad)
+                {
+                    diagnostics.Add(ErrorCode.ERR_PatternWrongType, expression.Syntax.Location, unionType, expression.Display);
+                    hasErrors = true;
```

Comment:
It doesn't look like `hasErrors` is used after being assigned here, was that intentional?

Reply:
> It doesn't look like hasErrors is used after being assigned here, was that intentional?

Yes, a copy/paste artifact. I'll remove this assignment (probably in the next PR)